automation/src/test/scala/org/broadinstitute/dsde/firecloud/api/FireCloudClient.scala

```diff
@@ -0,0 +1,113 @@
+package org.broadinstitute.dsde.firecloud.api
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpMethods._
+import akka.http.scaladsl.model.headers.{Authorization, OAuth2BearerToken}
+import akka.http.scaladsl.model.{Multipart, _}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Sink, _}
+import akka.util.ByteString
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.module.scala.DefaultScalaModule
+import org.broadinstitute.dsde.firecloud.config.AuthToken
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, ExecutionContext, ExecutionContextExecutor, Future}
+
+class FireCloudClient {
+  implicit val system = ActorSystem()
+  implicit val materializer = ActorMaterializer()
+  implicit val ec: ExecutionContextExecutor = ExecutionContext.global
+
+  val mapper = new ObjectMapper()
+  mapper.registerModule(DefaultScalaModule)
+
+  // TODO: Is this okay here or should it be in a separate object and explicitly imported?
+  implicit class JsonStringUtil(s: String) {
+    def fromJsonMapAs[A](key: String): Option[A] = parseJsonAsMap.get(key)
+
+    def parseJsonAsMap[A]: Map[String, A] = mapper.readValue(s, classOf[Map[String, A]])
+  }
+
+  private def makeAuthHeader(token: AuthToken): Authorization = {
+    headers.Authorization(OAuth2BearerToken(token.value))
+  }
+
+  private def sendRequest(httpRequest: HttpRequest): HttpResponse = {
+    Await.result(Http().singleRequest(httpRequest), 5.minutes)
+  }
+
+  private def handleResponse(response: HttpResponse): String = {
+    response.status.isSuccess() match {
+      case true =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        Await.result(entityFuture, 100.millis).decodeString("UTF-8")
+      case _ =>
+        val byteStringSink: Sink[ByteString, Future[ByteString]] = Sink.fold(ByteString("")) { (z, i) => z.concat(i) }
+        val entityFuture = response.entity.dataBytes.runWith(byteStringSink)
+        throw new APIException(Await.result(entityFuture, 100.millis).decodeString("UTF-8"))
+    }
+  }
+
+  def parseResponse(response: HttpResponse): String = {
```

Comment:
What's the difference between `handleResponse` and `parseResponse`? They look identical to me.

Also I feel like this could be achieved more simply by acting on futures instead of chaining multiple `Await.result` calls. My attempt (annotated):
```
 private def sendRequestAndWaitForResponse(httpRequest: HttpRequest): String = {
    val future = for {
      response <- Http().singleRequest(httpRequest)  // makes the request and returns a Future[HttpResponse]
      responseAsString <- response.entity.toStrict(5 seconds).map(_.data.decodeString("UTF-8"))  // Loads the response into memory and decodes it as a String, returning a Future[String]
    } yield {
      // Throw exception if the response is a failure, else return it
      if (response.status.isFailure()) {
        throw new APIException(s"Request failed with status ${response.status.intValue}: $s")
      } else {
        responseAsString
      }
    }
    // Either returns a string or throws an exception
    Await.result(future, 5 minutes)
  }
```